phoenix-core/src/it/java/org/apache/phoenix/end2end/BackwardCompatibilityTestUtil.java

```diff
@@ -0,0 +1,301 @@
+                if ( !(line.isEmpty() || line.startsWith("*") || line.startsWith("/"))) {
+                    goldFile.add(line);
+                }
+            }
+        }
+
+        // We take the first line in gold file and match against the result file to exclude any
+        // other WARNING messages that comes as a result of the query execution
+        int index = resultFile.indexOf(goldFile.get(0));
+        assertNotEquals("Mismatch found between gold file and result file", -1, index);
+        resultFile = resultFile.subList(index, resultFile.size());
+        assertEquals(goldFile, resultFile);
+    }
+
+    // Executes the SQL commands listed in the given operation file from the sql_files directory
+    public static void executeQueriesWithCurrentVersion(String operation, String url,
+                                                        boolean setMaxLookBackAge) throws Exception {
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        if (setMaxLookBackAge) {
+            props.put(
+                    QueryServices.GLOBAL_INDEX_ROW_AGE_THRESHOLD_TO_DELETE_MS_ATTRIB,
+                    Long.toString(0));
+            props.put(ScanInfoUtil.PHOENIX_MAX_LOOKBACK_AGE_CONF_KEY,
+                    Integer.toString(15));
+        }
+
+        try (Connection conn = DriverManager.getConnection(url, props)) {
+            StringBuilder sb = new StringBuilder();
+            try (BufferedReader reader =
+                         getBufferedReaderForResource(SQL_DIR + operation + SQL_EXTENSION)) {
+                String sqlCommand;
+                while ((sqlCommand = reader.readLine()) != null) {
+                    sqlCommand = sqlCommand.trim();
+                    if (sqlCommand.length() == 0 || sqlCommand.startsWith("/")
+                            || sqlCommand.startsWith("*")) {
+                        continue;
+                    }
+                    sb.append(sqlCommand);
+                }
+            }
+            ResultSet rs;
+            String[] sqlCommands = sb.toString().split(";");
+
+            URL fileUrl = BackwardCompatibilityIT.class.getClassLoader().getResource(
+                    RESULTS_AND_GOLD_FILES_DIR);
+            assertNotNull(fileUrl);
+            final String resultFile = new File(fileUrl.getFile()).getAbsolutePath() + "/" +
+                    RESULT_PREFIX + operation + TEXT_EXTENSION;
+            try (BufferedWriter br = new BufferedWriter(new FileWriter(resultFile))) {
+                for (String command : sqlCommands) {
+                    try (PreparedStatement stmt = conn.prepareStatement(command)) {
+                        stmt.execute();
+                        rs = stmt.getResultSet();
+                        if (rs != null) {
+                            saveResultSet(rs, br);
+                        }
+                    }
+                    conn.commit();
+                }
+            }
+        }
+    }
+}
```

Comment:
@yanxinyi my assumption is there is no _real_ diff in this class and is just a bunch of stuff copied from `BackwardCompatibilityIT` right?

Reply:
Yes, no logic changes. The only diff is adding extra param for zkQuorm or URL for the execute script.